Combine name and submitted_by fields into authored_by

biru-codeastromer opened the pull request:
### Fixes #4 

## Description:

This pull request addresses the following changes to the YAML files in the project:

1. ### Field Consolidation:
    - Initially combined `name` and `submitted_by` fields into a single `combined_name_submitted_by` field.
    - Resolved the issue where both `name` and `submitted_by` were concatenated resulting in duplicate names.

2. ### Field Update:
    - Replaced the `combined_name_submitted_by` field with a more appropriate `authored_by` field.
    - Ensured that the `authored_by` field correctly reflects the author's name without duplication.

3. ### Script Enhancement:
    - Updated the script to:
        - Check for the presence of both `combined_name_submitted_by` and `submitted_by` fields.
        - Consolidate these fields into the new `authored_by` field within the `metadata` section.
        - Ensure the old fields (`combined_name_submitted_by` and `submitted_by`) are removed after the consolidation.

## Rationale:

- ### Consistency: 
  - Having a single field (`authored_by`) improves the consistency of the data structure.
- ### Clarity: 
  - It clarifies the purpose of the field by using a more descriptive name.
- ### Maintenance: 
  - Simplifies the maintenance and future updates of the YAML files.

## Testing and Validation:
  - Verified that all relevant YAML files are updated correctly.
  - Ensured no data is lost during the field consolidation.
  - Checked for any potential issues in the parsing and writing of the YAML files.

Please review the changes and let me know if there are any questions or further adjustments needed.

krisstern commented on combineFields.js:
```diff
@@ -0,0 +1,57 @@
+const path = require('path');
+const yaml = require('yaml');
+
+function combineFields(filePath) {
```
Is this file for the automated changes? If so there is no need to include in the PR.

krisstern replied:
This is because this would be a one-time data change

biru-codeastromer replied:
Hi Sir,

Thank you for your feedback! I understand your point regarding the combineFields.js script being unnecessary for a one-time data change. I’ll try to implement the logic directly within gatsby-node.js itself to handle the field transformation during the build process.

I’ll update the PR soon and let you know once the changes are made.

krisstern reviewed (commented):
I will likely need some time to review this PR locally, as this is a rather "big" PR. So it may be days before I get back on this.

biru-codeastromer commented:
> I will likely need some time to review this PR locally, as this is a rather "big" PR. So it may be days before I get back on this.

Sure Sir!

krisstern commented on gatsby-node.js:
```diff
@@ -61,6 +61,36 @@ exports.onCreateNode = async ({node, actions, loadNodeContent, createNodeId, cre
+                    obj.metadata = {};
+                }
+
+                const mapFields = ['name', 'submitted_by', 'authored_by', 'location', 'industries', 'latitude', 'longitude'];
```
Regarding
```
                const mapFields = ['name', 'submitted_by', 'authored_by', 'location', 'industries', 'latitude', 'longitude'];
```
Why do we still have the "name" and "submitted_by" fields in this PR?

biru-codeastromer replied:
Hi Sir,

Thank you for pointing that out! The presence of the name and submitted_by fields in the PR is unintentional. I initially retained these fields during the migration to ensure no data was lost, but I missed cleaning them up in the final implementation.

I will update the PR to remove these redundant fields and ensure that only the authored_by field is used consistently across the codebase. 

Thank you for catching this, and I’ll notify you as soon as the changes are made.

biru-codeastromer replied:
I have made the changes and also tested it locally to check if everything runs fine !

<img width="1512" alt="Screenshot 2025-01-01 at 1 25 26 AM" src="https://github.com/user-attachments/assets/83957889-5cd0-44c2-be41-5f0db2dd296b" />

<img width="1512" alt="Screenshot 2025-01-01 at 1 25 44 AM" src="https://github.com/user-attachments/assets/1a14ee9c-a5fc-445f-a228-38ba4bf99bbb" />

<img width="1512" alt="Screenshot 2025-01-01 at 1 26 07 AM" src="https://github.com/user-attachments/assets/b3de4d5b-d0db-44be-9c11-bcc990b555c4" />

<img width="1512" alt="Screenshot 2025-01-01 at 1 28 52 AM" src="https://github.com/user-attachments/assets/cdb87ed4-a15f-433b-910d-903081947452" />

krisstern reviewed (changes requested):
Can you undo the 4-space changes back to 2-space please for the formatting?

biru-codeastromer commented:
> Can you undo the 4-space changes back to 2-space please for the formatting?

Hi Sir , I’ve reverted all 4-space indentation changes back to 2-space for better formatting .

krisstern reviewed (changes requested):
Please undo all the unrelated formatting changes in this PR

biru-codeastromer commented:
> Please undo all the unrelated formatting changes in this PR

Hi Sir,
Thank you for the feedback! I’ll review the entire PR and ensure that all unrelated formatting adjustments are reverted to align with the repo’s standards.

I’ll update the PR shortly and notify you once the changes are completed.

biru-codeastromer commented:
> Please undo all the unrelated formatting changes in this PR

Hi Sir, I made the indentation 4 spaces because I was encountering linting errors when using 2 spaces.

<img width="1512" alt="Screenshot 2025-01-02 at 1 29 19 PM" src="https://github.com/user-attachments/assets/711fca66-7b53-4dc8-aac2-143e21fa21e7" />

biru-codeastromer commented:
Regarding the other unrelated formatting changes in this PR, I am currently reviewing them and will revert those changes. I'll ensure that only the necessary code modifications are included.

krisstern commented on gatsby-node.js:
```diff
@@ -103,28 +122,49 @@ exports.onCreateNode = async ({node, actions, loadNodeContent, createNodeId, cre
+      summary: String
+      team_members: [String]
+      version_control_systems: [String]
+      authored_by: String
+      location: String
+      latitude: String
+      longitude: String
```
```suggestion
```

krisstern replied:
Why are these considered as a part of the `UserStoryMetadata` now? Any justification for it?

krisstern commented on gatsby-node.js:
```diff
@@ -103,28 +122,49 @@ exports.onCreateNode = async ({node, actions, loadNodeContent, createNodeId, cre
-          title: String
-          paragraphs: [MarkdownRemark] @link
-        }
+    type UserStoryBody_content @dontinfer {
+      title: String
+      paragraphs: [MarkdownRemark] @link
+    }
+
+    type UserStoryMap {
+      location: String
+      industries: [String]
+      authored_by: String
+      latitude: String
+      longitude: String
+    }
```
```suggestion
```

krisstern replied:
Ditto, why are these needed now but not before?

krisstern commented on gatsby-node.js:
```diff
@@ -103,28 +122,49 @@ exports.onCreateNode = async ({node, actions, loadNodeContent, createNodeId, cre
+      longitude: String
+    }
 
+    type UserStory implements Node {
+      id: ID!
+      slug: String!
+      metadata: UserStoryMetadata
+      body_content: UserStoryBody_content
+      map: UserStoryMap
+      next: UserStory
+      previous: UserStory
+    }
```
```suggestion
```

krisstern replied:
Same question here: Why are these needed here now?

krisstern reviewed (changes requested):
Can you get rid of the unused `"fs"` package from the "package.json" as well as "package-lock.json" files? If there were in use before please keep them as is.

Also, I am still seeing unrelated formatting changes in the files changes in this PR. Can you revert them?

biru-codeastromer commented:
Hi Sir,

Thank you for your detailed feedback. 

I will verify the usage of the fs package. If it is not required, I will remove it from both package.json and package-lock.json to keep dependencies clean. If it serves any purpose, I will ensure it remains properly documented.
Revert Unrelated Formatting Changes:

I will review the PR to identify and revert any formatting changes not related to the core purpose of the submission. This will ensure the changes stay focused and easier to review.


The purpose of replacing name and submitted_by with the single authored_by field is to enhance the consistency and clarity of the metadata structure. This approach reduces redundancy while maintaining all relevant information.

If the inclusion of any additional fields (like location, latitude, or longitude) is not crucial to the current requirements, I will ensure they are excluded or introduced only when there is a demonstrated need.


I  committed to removing these fields based on your suggestion but ran into some issues during the process. I will revisit this, properly examine the changes, and ensure the PR reflects the necessary updates. I will notify you once this is done.

Thank you for your guidance and patience.

biru-codeastromer commented:
> Can you get rid of the unused `"fs"` package from the "package.json" as well as "package-lock.json" files? If there were in use before please keep them as is.
> 
> Also, I am still seeing unrelated formatting changes in the files changes in this PR. Can you revert them?


Hi Sir,

I have addressed the requested changes as follows:

1.I have reviewed the usage of the fs package and confirmed that it is not in use. Consequently, I have removed it from both package.json and package-lock.json.

2.I have carefully reviewed the entire PR and reverted any unrelated formatting changes to ensure consistency with the repository's standards.

3.I have removed the location, latitude, and longitude fields from UserStoryMetadata, and retained only the authored_by field.
Similarly, I have removed the UserStorymap part from gatsby-node.js

Please review the updates and let me know if there are any further adjustments needed.

biru-codeastromer commented:
The authored_by field has been retained in UserStoryMetadata to ensure clear and consistent attribution of authorship for each user story. This approach eliminates redundancy, centralizes the author's information, and simplifies future updates and maintenance. Keeping authored_by within the metadata ensures that the data remains accessible, reliable, and easy to manage as the application evolves.

biru-codeastromer commented:
The authored_by field has been retained in UserStoryMetadata to ensure clear and consistent attribution of authorship for each user story. This approach eliminates redundancy, centralizes the author's information, and simplifies future updates and maintenance. Keeping authored_by within the metadata ensures that the data remains accessible, reliable, and easy to manage as the application evolves.

biru-codeastromer commented:
Regarding the following code snippet:

```
type UserStoryBody_content @dontinfer {
  title: String
  paragraphs: [MarkdownRemark] @link
}
```

I mistakenly removed this part of the code in my initial changes. However, after reviewing, I realized it was already present in the original code within the repository. Therefore, I have brought it back as part of my updates. 


<img width="1512" alt="Screenshot 2025-01-03 at 2 49 21 AM" src="https://github.com/user-attachments/assets/66ab08ec-dd39-4290-b2e4-3827d9db10e6" />

krisstern commented:
Could you please add the following to "devDependencies" in "package.json"?:
```console
"@babel/plugin-proposal-private-property-in-object": "^7.21.11",
```

Also, could you please update by running the following:
```console
npx browserslist@latest --update-db
```

Then, run linting by doing:
```console
npm run lint
```

Finally, I think this is the right time to introduce Prettier (https://prettier.io/) into the mix for more consistent file formatting. Currently the formatting is supposed to be automated with EditorConfig, but I don't think it is powerful enough to clear up the formatting inconsistencies that we are currently encountering. You may want to explore how to set this up to replace the need to have a ".editorconfig" file at root. Would you be interested in implementing such changes?

krisstern commented:
Also, according to the ".editorconfig" file we currently have:

```
# EditorConfig is awesome: http://EditorConfig.org
root = true

[*.{css,js,jsx,json}]
end_of_line = lf
insert_final_newline = true
charset = utf-8
indent_style = space
indent_size = 4

[package.json]
indent_size = 2

[plugins/**/package.json]
indent_size = 2
```

Could you please upgrade the formatting to having 4 spaces as indent for the `.css`, `.js`, `.jsx`, and `.json` files then, preferably using Prettier to automate the formatting?

krisstern commented on package.json:
```diff
@@ -42,9 +42,10 @@
   "devDependencies": {
-    "@jenkinsci/gatsby-plugin-jenkins-layout": "^1.5.0"
+    "@jenkinsci/gatsby-plugin-jenkins-layout": "^1.5.0",
+    "gh-pages": "^6.2.0"
```
```suggestion
```

krisstern replied:
What do we need this here for? I don't think we are hosting using GitHub Pages?

biru-codeastromer replied:
My bad, this was a mistake. I will remove the "gh-pages": "^6.2.0" dependency immediately.

Thank you for pointing that out!

biru-codeastromer commented:
Hi Sir,
Thanks for the suggestions! I have made all the changes .

-I have added  ``` "@babel/plugin-proposal-private-property-in-object": "^7.21.11" ``` to devDependencies and run npm install;
-Updated the browser list with ``` npx browserslist@latest --update-db``` ;
-Ran and checked the linting command with ``` npm run lint```.

krisstern commented:
How about upgrading from using EditorConfig to Prettier for formatting? Are you interested in add that to this PR as well?

biru-codeastromer commented:
> Finally, I think this is the right time to introduce Prettier (https://prettier.io/) into the mix for more consistent file formatting. Currently the formatting is supposed to be automated with EditorConfig, but I don't think it is powerful enough to clear up the formatting inconsistencies that we are currently encountering. You may want to explore how to set this up to replace the need to have a ".editorconfig" file at root. Would you be interested in implementing such changes?

I agree that introducing Prettier would help maintain consistent file formatting across the project. I’m definitely interested in implementing it and replacing the need for the .editorconfig file. I’ll explore how to set it up and ensure it's properly integrated for automated formatting.

Thanks for the suggestion!




> Could you please update the formatting to having 4 spaces as indent for the .css, .js, .jsx, and .json files then, preferably using Prettier to automate the formatting?

I’ll update the indentation for the .css, .js, .jsx, and .json files to use 4 spaces, as requested after setting up Prettier. I’ll also make sure to automate this formatting using Prettier to ensure consistency across the project.

biru-codeastromer commented:
> How about upgrading from using EditorConfig to Prettier for formatting? Are you interested in add that to this PR as well?

Definitely interested Sir as it would be much better !

biru-codeastromer commented:
> How about upgrading from using EditorConfig to Prettier for formatting? Are you interested in add that to this PR as well?

Done, Sir!
I have successfully introduced Prettier into the PR. It ran without any issues, and I have implemented the necessary formatting changes by running npm run prettier as required.

Please let me know if the codes in the prettier files needs to be adjusted.

Here's a screenshot of the applied changes:

<img width="1512" alt="Screenshot 2025-01-03 at 4 44 13 AM" src="https://github.com/user-attachments/assets/651d5f90-e2ca-48d5-834c-435cef0e3792" />

krisstern commented on package.json:
```diff
@@ -13,7 +13,9 @@
     "clean": "gatsby clean",
-    "lint": "eslint ."
+    "lint": "eslint .",
+    "prettier": "prettier --write \"src/**/*.{css,js,jsx,json}\"",
```
```suggestion
    "format": "prettier --write \"src/**/*.{css,js,jsx,json}\"",
```

krisstern commented on package.json:
```diff
@@ -13,7 +13,9 @@
-    "lint": "eslint ."
+    "lint": "eslint .",
+    "prettier": "prettier --write \"src/**/*.{css,js,jsx,json}\"",
+    "prettier-check": "prettier --check \"src/**/*.{css,js,jsx,json}\""
```
```suggestion
    "format-check": "prettier --check \"src/**/*.{css,js,jsx,json}\""
```

krisstern commented:
Also, to remove some of the fixable dependency vulnerabilities, could you please run the following and commit the changes?:
```console
npm audit fix
```

krisstern commented on .prettierrc:
```diff
@@ -0,0 +1,12 @@
+  "singleQuote": true,
+  "trailingComma": "all",
+  "bracketSpacing": true,
+  "jsxBracketSameLine": false,
```
```suggestion
  "bracketSameLine": false,
```

krisstern replied:
I have checked that `"jsxBracketSameLine"` is deprecated

krisstern commented:
Also, can we use a `".prettierignore"` file to ignore some files while formatting, like the `.yaml` files which are not any one of the `.css`, `.js`, `.jsx`, and `.json` files? For more info, please see: https://prettier.io/docs/en/ignore.html

biru-codeastromer commented:
> Also, can we use a `".prettierignore"` file to ignore some files while formatting, like the `.yaml` files which are not any one of the `.css`, `.js`, `.jsx`, and `.json` files? For more info, please see: https://prettier.io/docs/en/ignore.html

I’ve added a .prettierignore file to exclude .yaml, .png, and the static directory. Prettier now skips these files during formatting.

krisstern commented on .prettierignore:
```diff
@@ -0,0 +1,3 @@
+*.yaml
+*.png
+static
```
```suggestion
static

```

krisstern reviewed (changes requested):
Better to leave a blank line in the EOF for this type of file.

krisstern commented on src/user-story/a-stable-projects-life/index.yaml:
```diff
@@ -1,9 +1,7 @@
----
```
Hey why are the line with three dashes up top removed? It is needed in a standard YAML file format. According to https://www.cloudbees.com/blog/yaml-tutorial-everything-you-need-get-started, 
> The file starts with three dashes. These dashes indicate the start of a new YAML document. YAML supports multiple documents, and compliant parsers will recognize each set of dashes as the beginning of a new one.

biru-codeastromer replied:
Hi Sir,  I’m planning to revert the files to the earlier commit with the correct formatting of yaml files and update the Prettier configuration to preserve YAML-specific elements moving forward. Does this approach work for you?

biru-codeastromer replied:
Like instead of : 
```
{
  "printWidth": 80,
  "tabWidth": 4,
  "useTabs": false,
  "semi": true,
  "singleQuote": true,
  "trailingComma": "all",
  "bracketSpacing": true,
  "bracketSameLine": false,
  "arrowParens": "avoid",
  "endOfLine": "lf"
}
```

How about this : 
```
{
  "printWidth": 80,
  "tabWidth": 4,
  "useTabs": false,
  "semi": true,
  "singleQuote": true,
  "trailingComma": "all",
  "bracketSpacing": true,
  "jsxBracketSameLine": false,
  "arrowParens": "avoid",
  "endOfLine": "lf",
  "overrides": [
    {
      "files": "*.yaml",
      "options": {
        "proseWrap": "preserve"
      }
    }
  ]
}
```

krisstern replied:
You already have the `.prettierignore` set up, so the key is the apply the `.prettierignore` before you run `npm run format` after reverting everything to before you first apply the prettier formatting. I don't think the `"overrides" work this way TBH.

biru-codeastromer replied:
Sure Sir, going to do it and revert necessary changes!

biru-codeastromer replied:
Hi Sir,

I wanted to notify you about something I noticed. While I tried reverting to an older commit (before the Prettier-related changes), I don’t know why the dashes (---) didn’t show up even in that commit. The files should have had the correct format prior to the Prettier commit, but it seems the dashes were somehow removed permanently. I even checked in a different window in VS Code to confirm, but they still didn’t appear.

As a result, I manually corrected the YAML files to ensure the formatting is accurate. Will it cause any issues if I run ```npm run format``` again in the future?

Thank you!

krisstern replied:
Can you tell me more how you reverted the changes?

krisstern replied:
Did you use `git rebase ...` at all?

krisstern replied:
Or a variant of `git reset ...` or `git checkout ...` would do too, but I don't think you have used any of these since I can still see your supposedly reverted commits. (If they are reverted, they should not be in the commit history of this PR at all.)

biru-codeastromer replied:
Hi Sir,

Thank you for pointing this out. Let me clarify:

I initially tried reverting the changes by checking out an older commit using git checkout <commit-hash> to a point before the Prettier changes. However, the dashes (---) were not restored, even in that older commit. I confirmed this by inspecting the files in VS Code.

I did not use git rebase or git reset; my approach was limited to git checkout. As a result, the previous commits are still visible in the PR history.

After realizing the older commits did not restore the formatting, I manually corrected the YAML files to include the dashes (---). I used VS Code's search-and-replace functionality to make these corrections efficiently.

Will these manual changes cause any issues going forward?

Thank you for your guidance!

biru-codeastromer replied:
Hi Sir,

I wanted to update you regarding the recent changes. I tested the setup by running npm install and npm run format to check if the YAML files would be affected again. Thankfully, the .prettierignore file seems to be working as expected this time, and the YAML files were not reformatted.

<img width="1512" alt="Screenshot 2025-01-05 at 10 14 38 PM" src="https://github.com/user-attachments/assets/a66164a8-c6f5-4a2d-8899-52b60f38594a" />
<img width="1512" alt="Screenshot 2025-01-05 at 10 13 57 PM" src="https://github.com/user-attachments/assets/d63105c3-f1ca-4a79-ac70-90f193360d41" />


This confirms that the dashes (---) I manually restored remain intact. Everything appears to be functioning properly now. Please let me know if there’s anything else I should verify or address.

Thank you!

krisstern replied:
If you use `git checkout <commit-hash>` and it does not work, then you should try `git reset --hard <commit-hash>` instead. If I still see your previous commits about prettier in the PR commit history it means you are not doing something right. I know this may be a bit much to ask of you, but would you like to **first back up your prettier (file as well as code) changes external to the repo** and then **give `git reset --hard <commit-hash>` a try**? I am making this suggestion because I noticed some other unwanted YAML file changes were introduced because you were not aware you needed a `.prettierignore` file before.

biru-codeastromer replied:
Hi Sir,

Thank you for the clarification and suggestion. I understand where I might have gone wrong earlier. I will back up my current changes (including the Prettier-related updates and manually corrected YAML files) outside the repository for safekeeping.

Then, I’ll try using ``` git reset --hard <commit-hash> ``` to ensure the commit history is reverted correctly and no unwanted changes remain. I’ll also verify that the ```.prettierignore``` file is applied properly before proceeding with any formatting commands to avoid reintroducing issues.

I appreciate your patience and guidance on this. I’ll let you know once I’ve completed the steps and tested everything again.

Thank you!

biru-codeastromer replied:
<img width="1512" alt="Screenshot 2025-01-06 at 12 22 05 AM" src="https://github.com/user-attachments/assets/df65a6a0-de93-45f3-aecc-452f24e9c5d5" />


Hi Sir,

I followed your suggestion to use git reset --hard 21a7082 to revert the changes and checked the YAML files afterward. However, I noticed that the --- dashes at the beginning of the YAML files were still missing.

To confirm this, I inspected the state of the files in commit 21a7082 using git show, and it seems that the --- dashes were not present in that commit either. It appears that the issue originated earlier and wasn’t part of the changes introduced after that commit.

Would you recommend manually restoring the --- dashes to the YAML files now and ensuring they are preserved going forward with a .prettierignore file?

Thank you for your guidance!

biru-codeastromer replied:
Also Sir , would you recommend bringing back the backup files I created, manually restoring the --- dashes to the YAML files, and then pushing the changes using git push --force? This would ensure the commit history is fresh with the previous prettier commits getting removed, but the YAML files will have the corrections applied manually as per my backup.

krisstern replied:
Which IDE are you using for the local development?

biru-codeastromer replied:
Hi Sir,

Thank you for your response. I am currently using VS Code as my IDE for local development.

krisstern replied:
Yeah, that may be why... Have you configured your VS Code IDE to autosave on every change?

biru-codeastromer replied:
Hi Sir,

Thank you for your guidance. Upon checking my VS Code settings, I confirmed that auto-save is turned off in my IDE. 

<img width="1512" alt="Screenshot 2025-01-06 at 11 26 37 PM" src="https://github.com/user-attachments/assets/4d5438f3-becc-4304-9f8e-afe5c5dda799" />

krisstern replied:
You may want to turn autosave on and try again

biru-codeastromer replied:
Hi Sir,

Thank you for the suggestion. I will turn on auto-save in my IDE settings and try again. I'll then re-check the YAML files and ensure the --- dashes are preserved after making changes.

Thank you!

biru-codeastromer replied:
> You may want to turn autosave on and try again

<img width="1512" alt="Screenshot 2025-01-07 at 12 00 43 AM" src="https://github.com/user-attachments/assets/0f829084-8e25-4c41-89db-29581e27191c" />

Hi Sir,

I’ve been troubleshooting an issue where autosave is enabled and I’ve been switching between commits. After enabling autosave and updating to the latest commit, I reverted to the commit before Prettier was introduced in order to inspect the YAML files. However, I noticed that the three dashes (---) at the beginning of the YAML files are still missing.

I’m uncertain whether the problem is related to my IDE or something else.

I do have a backup of the files with the correct formatting, and I’m considering restoring them from the backup. Would you recommend this approach, or is there a better way to resolve the issue? My plan is to then re-run everything, ensuring Prettier works as expected without altering any files, and perform a ```git push --force``` to push the changes.

Thank you for your guidance!

krisstern replied:
I don't exactly know what's wrong with your setup, but definitely something is wrong, and I don't know everything about your setup and you are not providing very useful information. I am not supposed to be doing debugging for your local setup, so you are better off trying a bit harder on your side to find out what needs fixing.

krisstern replied:
Alternatively you can also try using something like `git rebase -i HEAD~20` to interactive **drop** commits from your PR history.

biru-codeastromer replied:
> I don't exactly know what's wrong with your setup, but definitely something is wrong, and I don't know everything about your setup and you are not providing very useful information. I am not supposed to be doing debugging for your local setup, so you are better off trying a bit harder on your side to find out what needs fixing.



Hi Sir,

Thank you for your feedback. I understand the importance of troubleshooting this on my own, and I’ll investigate further to identify the issue with my setup .
If possible will try with other IDE.

biru-codeastromer replied:
> Alternatively you can also try using something like `git rebase -i HEAD~20` to interactive **drop** commits from your PR history.


I will also try using `git rebase -i HEAD~20` to clean up the PR history and ensure that only the relevant commits remain.

krisstern reviewed (changes requested):
Please revert all of the removals of the beginning three dashes.

biru-codeastromer commented:
> Please revert all of the removals of the beginning three dashes.

Hi Sir,

I have manually reverted all the YAML files to include the three dashes (---) at the top. While I attempted to revert to an older commit to restore the formatting automatically, it didn’t work as expected. As a result, I completed the corrections manually to ensure the files are accurate.

krisstern commented:
To be honest I don't think you are supposed to revert the changes to the YAML files manually

biru-codeastromer commented:
> To be honest I don't think you are supposed to revert the changes to the YAML files manually

I understand your concern about not manually reverting the changes to the YAML files. I will research this further to identify the best approach to resolve the issue correctly without relying on manual adjustments.

I appreciate your guidance and will update you once I have a more robust solution in place.

biru-codeastromer commented:
> To be honest I don't think you are supposed to revert the changes to the YAML files manually

It seems like the --- in the YAML files might have been removed in a subsequent commit. Do you think it would be a good idea to roll back to the previous commit where Prettier was set up, then reapply the setup? After that, we could use the following command:
``` find src/user-story/ -name '*.yaml' -exec sed -i '' '1s/^/---\n/' {} +```
This approach should restore the --- dashes at the beginning of each YAML file. Do you think this method is suitable, Sir?

biru-codeastromer commented:
> > To be honest I don't think you are supposed to revert the changes to the YAML files manually
> 
> It seems like the --- in the YAML files might have been removed in a subsequent commit more prior to prettier. Do you think it would be a good idea to roll back to the previous commit where Prettier was set up, then reapply the setup? After that, we could use the following command: ` find src/user-story/ -name '*.yaml' -exec sed -i '' '1s/^/---\n/' {} +` This approach should restore the --- dashes at the beginning of each YAML file. Do you think this method is suitable, Sir?

biru-codeastromer commented:
I will thoroughly debug the issue further to ensure that YAML files' first lines remain unchanged in any commit, as I understand your preference to avoid automated modifications for adding the three dashes (---). I’ll take the necessary time to analyze and resolve this properly, avoiding unintended changes. I will keep you updated here with my progress.

Thank you for your patience and understanding!

krisstern commented:
One hint is that if your old commit is still present or if you have additional commits in your commit history it is telling you that you are not really reverting the changes.

biru-codeastromer commented:
> > > To be honest I don't think you are supposed to revert the changes to the YAML files manually
> > 
> > 
> > It seems like the --- in the YAML files might have been removed in a subsequent commit more prior to prettier commit. Do you think it would be a good idea to roll back to the previous commit where Prettier was set up, then reapply the setup? After that, we could use the following command: ` find src/user-story/ -name '*.yaml' -exec sed -i '' '1s/^/---\n/' {} +` This approach should restore the --- dashes at the beginning of each YAML file. Do you think this method is suitable, Sir?
> 
> It is because you are not really reverting the changes





Hi Sir,

I have reverted the files to a commit where the issue did not exist. Now, I will carefully adjust the other files to incorporate the latest changes and ensure everything is aligned perfectly.

Thank you for your guidance, and I will keep you updated as I proceed further.

biru-codeastromer commented:
Hi Sir,

I wanted to inform you about the recent progress. After reverting the YAML files, I made all the necessary updates. However, I mistakenly used force push, which caused issues with the commits.

To resolve this, I went back to the state before the force push and will reapply the changes carefully, ensuring I avoid using force push this time. I’ll keep you updated on further progress.

Thank you for your understanding.

biru-codeastromer commented:
> It is because you are not really reverting the changes

Hi Sir, 
Finally I have solved all those issues as requested now :
-reverted yaml files to have those (---) dashes at top from previous commit.
-fixed the authored_by issue since it got erased in some files.

Please review it and let me know for further improvements.

krisstern commented on src/user-story/a-stable-projects-life/index.yaml:
```diff
@@ -62,11 +62,9 @@ body_content:
       *   issues can be highlighted, discovered, and remedied quickly
 title: Jenkins is the way to a stable project's life
 date: 2021-01-03T13:38:08.000Z
-post_name: a-stable-projects-life
```
Why are the `post_name` entries deleted from the YAML files?

biru-codeastromer replied:
Hi  Sir , 
I hope you are doing well . Please let me know if this issue has now been resolved.Thanks!

krisstern reviewed (changes requested):
Please restore all the `post_name` fields in the YAML files

biru-codeastromer commented:
> Please restore all the `post_name` fields in the YAML files

Hello Sir
I have done the restoration and also brought back the field in `_user_story.jsx`.
Please review it and let me know if further enhancements required.
Thanks!

biru-codeastromer commented:
> Please restore all the `post_name` fields in the YAML files

Hi Sir, I also want to ask about the version of yaml in `package.json` Should I keep it to the original repo version or should it be upgraded from `"yaml": "^2.0.0"` to latest version which is `2.7.0` .

krisstern commented on .prettierrc:
```diff
@@ -0,0 +1,12 @@
+{
+  "printWidth": 80,
+  "tabWidth": 4,
```
```suggestion
  "tabWidth": 2,
```

krisstern commented on src/components/UserStory.jsx:
```diff
@@ -62,34 +68,47 @@ const UserStory = ({image, title, submitted_by, tag_line, quotes, metadata, body
-                    </div>
+                    {metadata.authored_by && (
+                        <div className="container pt-2 pb-2">
+                            <strong>Authored By: {metadata.authored_by}</strong>
```
```suggestion
                            <strong>Authored By Jenkins User: {metadata.authored_by}</strong>
```

krisstern commented on src/components/UserStory.jsx:
```diff
@@ -125,23 +159,27 @@ UserStory.displayName = 'UserStory';
+            PropTypes.shape({
+                html: PropTypes.string.isRequired,
+            }),
+        ).isRequired,
```
Can you please explain why the extra `.isRequired` here?

krisstern commented on src/components/UserStory.jsx:
```diff
@@ -125,23 +159,27 @@ UserStory.displayName = 'UserStory';
+                html: PropTypes.string.isRequired,
+            }),
+        ).isRequired,
+    }).isRequired,
```
Same as above: Can you please explain why the extra `.isRequired` here?

biru-codeastromer replied:
Thank you Sir for your feedback. I initially added the .isRequired to the paragraphs array and the body_content object to enforce stricter validation, ensuring that these fields are always present and correctly structured. However, I realize now that this additional validation is not necessary and does not provide any additional benefit.

I will remove the extra .isRequired and keep the original structure intact.

I will make these adjustments immediately and update the PR shortly.

Thank you for your patience and guidance.

krisstern commented on src/pages/index.jsx:
```diff
@@ -1,26 +1,32 @@
+                        date
+                        tag_line
+                        slug
+                        metadata {
+                            authored_by
+                        }
```
Why is the `authored_by` wrapped around in the `metadata`? The original `submitted_by` is not I have checked?

biru-codeastromer replied:
Hi @krisstern,

Thank you for your feedback.

The authored_by field was wrapped in metadata because I included it in `UserStoryMetadata` and thought it was needed here for consistency.

biru-codeastromer replied:
If you feel my opinion is not vaild here, I will take your suggestion and adjust accordingly.
Please let me know your suggestions . 
Thanks!

biru-codeastromer commented:
I have removed the unwanted `isRequired` from `UserStory.jsx`.

krisstern commented on .prettierrc:
```diff
@@ -0,0 +1,12 @@
+{
+  "printWidth": 80,
+  "tabWidth": 2,
```
Could you please run `npm run format` again after having changed this to `2`?

biru-codeastromer replied:
Sure Sir!

krisstern reviewed (changes requested):
Can you please remove the `authored_by` instances from the metadata? Why is this field in particular a part of metadata?

krisstern reviewed (changes requested):
You will need to move all instances of the `authored_by` field outside of the `metadata` in all of the YAML files.

krisstern commented on gatsby-node.js:
```diff
@@ -61,6 +61,17 @@ exports.onCreateNode = async ({node, actions, loadNodeContent, createNodeId, cre
+                }
+            }
+
+            if (obj.metadata.authored_by) {
+                const uniqueNames = new Set(obj.metadata.authored_by.split(', ').map(name => name.trim()));
+                obj.metadata.authored_by = Array.from(uniqueNames).join(', ');
+            }
```
You will need to remove this and maybe replace with something appropriate since `authored_by` should not a part of metadata

biru-codeastromer replied:
Okay Sir,
Thanks for your detailed feedback . I will update all the requested changes and inform you about the update .

biru-codeastromer commented:
> You will need to move all instances of the `authored_by` field outside of the `metadata` in all of the YAML files.

Sure Sir ,updating it soon !

biru-codeastromer commented:
> Can you please remove the `authored_by` instances from the metadata? Why is this field in particular a part of metadata?

Hi Sir,
I have made all the changes as requested by you . Please review it and let me know if any improvement needed on it.
Thanks !

biru-codeastromer commented on gatsby-node.js:
```diff
@@ -61,8 +61,18 @@ exports.onCreateNode = async ({node, actions, loadNodeContent, createNodeId, cre
+                }
+            }
+
+            const authoredBy = obj.metadata?.authored_by || null;
+            delete obj.metadata?.authored_by;
```
Sir, I tried to delete the  authored by from metadata. I will remove since it doesnt did good to you Sir

krisstern replied:
Do you have any evidence we have had some similar lines of code for `submitted_by` or `name` which gave rise to these? Or are they coming out of nowhere?

biru-codeastromer commented on gatsby-node.js:
```diff
@@ -124,6 +134,12 @@ exports.createSchemaCustomization = ({actions: {createTypes}}) => {
           title: String
           paragraphs: [MarkdownRemark] @link
         }
+        type UserStoryMap {
+          authored_by: String
+        }
+        type UserStory implements Node {
+          authored_by: String
+        }
```
Actually Sir, after removing and doing your suggestion, I got error in `gatsby build` in which it was told to include it . So i implemented these two for those errors !

krisstern replied:
I am not suggesting anything I am asking why are these needed here

biru-codeastromer replied:
> I am not suggesting anything I am asking why are these needed here

Sir , after the removing `authored_by` from metadata i got an error in `gatsby build` regarding this that `authored_by` is not present in UserStoryMap and UserStory. That led me to use this to fix those issue in build

krisstern commented on gatsby-node.js:
```diff
@@ -61,8 +61,18 @@ exports.onCreateNode = async ({node, actions, loadNodeContent, createNodeId, cre
+
             const yamlNode = {
                 ...obj,
+                authored_by: authoredBy,
```
Why do we have this line of code and will the same errors from the lines added below go away if this is not here? By the lines below I meant the code snippets starting from https://github.com/jenkins-infra/stories/blob/a3ee8e2b7e4775bf21bd6c2312abbe36bd98348a/gatsby-node.js#L137.

biru-codeastromer replied:
I will test whether removing the line authored_by: authoredBy and its related schema lines resolves the issues without reintroducing errors during the build. I'll provide a detailed update after verifying this approach.

biru-codeastromer commented:
Thank you very much for all your constructive feedback, Sir! I truly appreciate your guidance. I will revisit the implementation to address the concerns raised and ensure it aligns with the requirements. This time, I will share the approach I plan to use with you before pushing any further changes. Thank you again for your patience and support!

biru-codeastromer commented:
< I don't understand what you are doing here, could you please explain? You are supposed to only combine and replace both name and submitted_by with authored_by, and somehow the changes you are proposing just don't make any sense to me. Is there any reason why?




Hi Sir,

I have finally fixed everything, and the unnecessary lines of code I previously included to address the issue have been removed. Upon reviewing my approach, I realized that I made a mistake by running `gatsby build` before properly moving `authored_by` out of metadata in all the 'yaml' files. This sequence caused errors during the `gatsby build` process, leading to the addition of those unnecessary lines from my side.

Now, I have resolved all the issues and ensured the changes align with the intended requirements. I hope you find the updated PR more aligned with your expectations. Please let me know if there are any further improvements or adjustments needed.

Thank you for your patience and feedback throughout this process!

biru-codeastromer commented:
I also want to mention that your suggestions played a significant role in refining the PR. Thank you for pointing out the necessary improvements Sir .

krisstern commented:
@halkeye Will we need any upstream changes to the database / CMS for this PR to work?

halkeye commented:
There's no upstream. Everything is files in this repo.

Also I would recommend splitting up this pr. One that adds prettier and mostly has whitespace changes. And one that has functionality changes. Doing both at the same time is a reviewing nightmare.

biru-codeastromer commented:
So should I remove the prettier part from this PR Sir @krisstern ?   as mentioned to create new PR.
Please let me know your feedback regarding this .

krisstern commented:
Hi @biru-codeastromer,
Yes, please go ahead to split this PR into a prettier part first which we will review first then the combining of the `name` and `submitted_by` fields into the new `authored_by` field.

biru-codeastromer commented:
> Hi @biru-codeastromer, Yes, please go ahead to split this PR into a prettier part first which we will review first then the combining of the `name` and `submitted_by` fields into the new `authored_by` field.



Sure Sir, also should I create create a new issue titled "upgrade eslint to prettier" and then the pr for that ? Please tell me if it's a good idea. 
Thanks

biru-codeastromer commented:
> > Hi @biru-codeastromer, Yes, please go ahead to split this PR into a prettier part first which we will review first then the combining of the `name` and `submitted_by` fields into the new `authored_by` field.
> 
> Sure Sir, also should I create create a new issue titled "upgrade eslint to prettier" and then the pr for that ? Please tell me if it's a good idea. Thanks!

krisstern commented:
> Sure Sir, also should I create create a new issue titled "upgrade eslint to prettier" and then the pr for that ? Please tell me if it's a good idea. Thanks !

We are not upgrading `eslint` to `prettier`, we are adding `prettier`, so you could call that PR "feat(dep): add prettier for formatting", following the convention as described at https://www.conventionalcommits.org/en/v1.0.0/.

biru-codeastromer commented:
> We are not upgrading `eslint` to `prettier`, we are adding `prettier`, so you could call that PR "feat(dep): add prettier for formatting", following the convention as described at https://www.conventionalcommits.org/en/v1.0.0/.


Thanks Sir !

biru-codeastromer commented:
> We are not upgrading `eslint` to `prettier`, we are adding `prettier`, so you could call that PR "feat(dep): add prettier for formatting", following the convention as described at https://www.conventionalcommits.org/en/v1.0.0/.


Hi Sir , I have created a separate PR #85 for the prettier part.
I have informed you about the update. Thanks !

biru-codeastromer commented:
Hi Sir @krisstern,

As PR #85 has been successfully merged, should I proceed to remove the Prettier-related changes from this PR and retain only the functional updates? Alternatively, would you recommend removing the Prettier changes and then pulling the latest changes from the main branch into this PR

Please let me know how you’d like me to proceed.

Thanks!

biru-codeastromer commented:
I think the conflicts came due to the latest changes in the main , I will fix it together .

krisstern commented:
> Hi Sir @krisstern,
> 
> As PR #85 has been successfully merged, should I proceed to remove the Prettier-related changes from this PR and retain only the functional updates? Alternatively, would you recommend removing the Prettier changes and then pulling the latest changes from the main branch into this PR
> 
> Please let me know how you’d like me to proceed.
> 
> Thanks!

Yes, please remove the prettier changes in this PR. You don't need to obtain permission to do this.

biru-codeastromer commented:
> Yes, please remove the prettier changes in this PR. You don't need to obtain permission to do this.


Hello Sir,
I have removed the prettier part from this PR completely now ! .
Please review the changes and let me know for further requirements .
Thanks .

krisstern commented on src/components/UserStory.jsx:
```diff
@@ -21,6 +21,22 @@ const titles = {
   team_members: 'Team Members',
   teams: 'Team',
   version_control_systems: 'Version Control System',
+  build_tools: 'Build Tools',
+  community_supports: 'Community Support',
+  company: 'Company',
+  company_website: 'Company Website',
+  funded_by: 'Funded By',
+  industries: 'Industries',
+  organization: 'Organization',
+  platforms: 'Platform',
+  plugins: 'Custom Plugins',
+  programming_languages: 'Programming Languages',
+  project_funding: 'Project Funding',
+  project_website: 'Project Website',
+  summary: 'Summary',
+  team_members: 'Team Members',
+  teams: 'Team',
+  version_control_systems: 'Version Control System',
```
I don't see these in the UserStory.jsx file before this PR, where is the idea for adding these coming from?

biru-codeastromer replied:
Hi Sir those were present in the main codebase itself : 

<img width="1512" alt="Screenshot 2025-01-18 at 8 10 14 PM" src="https://github.com/user-attachments/assets/223e7c20-2848-4d85-b8c1-63404a7f232a" />

krisstern commented on src/components/UserStory.jsx:
```diff
@@ -40,12 +56,28 @@ const fields = [
   'build_tools',
   'plugins',
   'community_supports',
+  'organization',
+  'company',
+  'company_website',
+  'teams',
+  'team_members',
+  'project_website',
+  'project_funding',
+  'funded_by',
+  'summary',
+  'industries',
+  'programming_languages',
+  'platforms',
+  'version_control_systems',
+  'build_tools',
+  'plugins',
+  'community_supports',
```
Same as above: I don't see these in the UserStory.jsx file before this PR, where is the idea for adding these coming from?

biru-codeastromer replied:
Hi Sir same it was present in the original repo files too : 

<img width="1512" alt="Screenshot 2025-01-18 at 8 10 14 PM" src="https://github.com/user-attachments/assets/1ebe1324-a06c-4d09-a4b2-784aa346b49c" />

krisstern commented on src/components/UserStory.jsx:
```diff
@@ -140,7 +174,7 @@ UserStory.displayName = 'UserStory';
   // children: PropTypes.node.isRequired,
   title: PropTypes.string.isRequired,
-  submitted_by: PropTypes.string.isRequired,
+  authored_by: PropTypes.string,
```
```suggestion
  authored_by: PropTypes.string.isRequired,
```

krisstern replied:
Please preserve what is in the original

biru-codeastromer replied:
Sure Sir thanks for the feedback !

krisstern commented on src/pages/_user_story.jsx:
```diff
@@ -75,6 +74,19 @@ UserStoryPage.propTypes = {
   data: PropTypes.shape({
     userStory: PropTypes.object.isRequired,
   }),
+  pageContext: PropTypes.shape({
+    previous: PropTypes.shape({
+      title: PropTypes.string.isRequired,
+      slug: PropTypes.string.isRequired,
+    }),
+    next: PropTypes.shape({
+      title: PropTypes.string.isRequired,
+      slug: PropTypes.string.isRequired,
+    }),
+  }),
+  data: PropTypes.shape({
+    userStory: PropTypes.object.isRequired,
+  }),
```
I don't see these in the UserStory.jsx file before this PR, where is the idea for adding these coming from?

krisstern commented on src/components/UserStory.jsx:
```diff
@@ -21,6 +21,22 @@ const titles = {
   team_members: 'Team Members',
   teams: 'Team',
   version_control_systems: 'Version Control System',
+  build_tools: 'Build Tools',
+  community_supports: 'Community Support',
+  company: 'Company',
+  company_website: 'Company Website',
+  funded_by: 'Funded By',
+  industries: 'Industries',
+  organization: 'Organization',
+  platforms: 'Platform',
+  plugins: 'Custom Plugins',
+  programming_languages: 'Programming Languages',
+  project_funding: 'Project Funding',
+  project_website: 'Project Website',
+  summary: 'Summary',
+  team_members: 'Team Members',
+  teams: 'Team',
+  version_control_systems: 'Version Control System',
```
```suggestion
```

krisstern replied:
It appears that they are already present in this same file in L8-23. Is there any reason for these duplicates?

krisstern replied:
Could you please commit this change @biru-codeastromer?

krisstern commented on src/components/UserStory.jsx:
```diff
@@ -40,12 +56,28 @@ const fields = [
   'build_tools',
   'plugins',
   'community_supports',
+  'organization',
+  'company',
+  'company_website',
+  'teams',
+  'team_members',
+  'project_website',
+  'project_funding',
+  'funded_by',
+  'summary',
+  'industries',
+  'programming_languages',
+  'platforms',
+  'version_control_systems',
+  'build_tools',
+  'plugins',
+  'community_supports',
```
```suggestion
```

Again, these are already present in L43-58. Is there any reason for their duplication?

biru-codeastromer replied:
No Sir .  The duplicate executed in trying to resolve conflicts . I will be refining it and inform you ! Thanks .

biru-codeastromer replied:
Hi Sir, I removed the duplicates from these two files thanks for your valuable updates !

krisstern replied:
Sams for this one

halkeye commented:
I don't know what the normal merge process is like but this pr now has 50+ commits.

Kris can you make sure to squash merge, or biru can you make sure you rebate squash all the commits.

biru-codeastromer commented:
> I don't know what the normal merge process is like but this pr now has 50+ commits.
> 
> Kris can you make sure to squash merge, or biru can you make sure you rebate squash all the commits.




Okay Sir , Sure .

biru-codeastromer commented:
> I don't know what the normal merge process is like but this pr now has 50+ commits.
> 
> Kris can you make sure to squash merge, or biru can you make sure you rebase squash all the commits.



Hi Sir @krisstern, 
As per HalkEye's sir suggestion, should I squash the commits myself and update the branch, or will you handle it via "Squash and Merge"? Please let me know how you'd like me to proceed.

krisstern commented:
> > I don't know what the normal merge process is like but this pr now has 50+ commits.
> > Kris can you make sure to squash merge, or biru can you make sure you rebase squash all the commits.
> 
> Hi Sir @krisstern, As per HalkEye's sir suggestion, should I squash the commits myself and update the branch, or will you handle it via "Squash and Merge"? Please let me know how you'd like me to proceed.

HI @biru-codeastromer did you not just said
> Okay Sir , Sure .
meaning you will rebase squash your commits?

biru-codeastromer commented:
No Sir @krisstern  I didnt mean that . But then I try it and do and update you here. 
Thanks!

biru-codeastromer commented:
Hi Sir's @krisstern and @halkeye  , I have rebased squash all commits into one  and fixed the conflicts and issues too ! 
Waiting for your review and feedback . Thanks !

krisstern commented:
Hi, I will review this PR shortly. But please refrain from pinging individuals unnecessarily unless it is absolutely needed to do so.

biru-codeastromer commented:
> Hi, I will review this PR shortly. But please refrain from pinging individuals unnecessarily unless it is absolutely needed to do so.

Thanks Sir,

Okay Sir I will always keep that in mind moving forward .
Sorry for this Sir , Will make sure it wont happen again .

krisstern reviewed (changes requested):
Just some final touches before this can be merged

biru-codeastromer commented:
> Just some final touches before this can be merged

I have done the changes Sir,
Thanks for the feedback and I removed the duplicates yet again . Sorry for it too !

krisstern reviewed (approved):
Thanks for all your hard work @biru-codeastromer! LGTM

biru-codeastromer commented:
> Thanks for all your hard work @biru-codeastromer! LGTM

Thank you so much Sir ,
I'm thrilled to see the PR merged and deeply appreciate your guidance and feedback throughout the process. It was a great learning experience for me, and I'm looking forward to contributing more  in the future. Thanks again for the support! 😊
